static/app/views/insights/common/components/widgets/overviewSlowAssetsWidget.tsx

```diff
@@ -0,0 +1,191 @@
+import {ModuleName, SpanFields} from 'sentry/views/insights/types';
+import {WidgetEmptyStateWarning} from 'sentry/views/performance/landing/widgets/components/selectableList';
+
+export default function OverviewAssetsByTimeSpentWidget(props: LoadableChartWidgetProps) {
+  const theme = useTheme();
+  const {query} = useTransactionNameQuery();
+  const releaseBubbleProps = useReleaseBubbleProps(props);
+  const {selection} = usePageFilters();
+  const organization = useOrganization();
+  const project = useAlertsProject();
+
+  const resourceQuery = getResourcesEventViewQuery({}, DEFAULT_RESOURCE_TYPES).join(' ');
+  const search = new MutableSearch(`has:span.group ${resourceQuery} ${query}`);
+  const referrer = Referrer.ASSETS_BY_TIME_SPENT;
+  const groupBy = SpanFields.NORMALIZED_DESCRIPTION;
+  const yAxes = 'sum(span.self_time)';
+  const totalTimeField = 'sum(span.self_time)';
+
+  const {
+    data: assetListData,
+    isLoading: isAssetListLoading,
+    error: assetListError,
+  } = useSpans(
+    {
+      fields: [
+        'span.group',
+        'project.id',
+        'sentry.normalized_description',
+        totalTimeField,
+      ],
+      sorts: [{field: totalTimeField, kind: 'desc'}],
+      search,
+      limit: 3,
+      noPagination: true,
+    },
+    referrer
+  );
+
+  const {
+    data: assetSeriesData,
+    isLoading: isAssetSeriesLoading,
+    error: assetSeriesError,
+  } = useTopNSpanSeries(
+    {
+      search: `span.group:[${assetListData?.map(item => `"${item['span.group']}"`).join(',')}]`,
+      fields: [groupBy, yAxes],
+      yAxis: [yAxes],
+      sort: {field: yAxes, kind: 'desc'},
+      topN: 3,
+      enabled: assetListData?.length > 0,
+    },
+    referrer
+  );
+
+  const isLoading = isAssetSeriesLoading || isAssetListLoading;
+  const error = assetSeriesError || assetListError;
+
+  const hasData = assetListData && assetListData.length > 0 && assetSeriesData.length > 0;
+
+  const colorPalette = theme.chart.getColorPalette(assetSeriesData.length - 1);
+
+  const visualization = (
+    <WidgetVisualizationStates
+      isEmpty={!hasData}
+      isLoading={isLoading}
+      error={error}
+      emptyMessage={<WidgetEmptyStateWarning />}
+      VisualizationType={TimeSeriesWidgetVisualization}
+      visualizationProps={{
+        id: 'overviewSlowAssetsWidget',
+        showLegend: props.loaderSource === 'releases-drawer' ? 'auto' : 'never',
+        plottables: assetSeriesData.map(
+          (ts, index) =>
+            new Line(convertSeriesToTimeseries(ts), {
+              color: colorPalette[index],
+            })
+        ),
+        ...props,
+        ...releaseBubbleProps,
+      }}
+    />
+  );
+
+  const footer = hasData && (
+    <WidgetFooterTable>
+      {assetListData?.map((item, index) => (
+        <Fragment
+          key={`${item['project.id']}-${item['span.group']}-${item['sentry.normalized_description']}`}
```

Comment:
Should we be using the `SpanFields` enum for all of these string literals that we're querying?

Reply:
I think technically this is still type same, but I like the idea of using span fields enum everywhere to easy code search. I'll update this!